includes/class-wc-stripe-api.php

```diff
@@ -242,6 +263,18 @@ public static function retrieve( $api ) {
 			]
 		);
 
+		// If we get a 401 error, we know the secret key is not valid.
+		if ( is_array( $response ) && ! empty( $response['response']['code'] ) && 401 === $response['response']['code'] ) {
```

Comment:
Minor thought: my code in #4327 is extremely defensive in this check, just to make sure that we have all the data we're expecting. Not blocking, but it may be worth being super-defensive to make sure we don't break in unexpected ways.
```suggestion
		if ( is_array( $response ) && isset( $response['response'] ) && is_array( $response['response'] ) && isset( $response['response']['code'] ) && 401 === $response['response']['code'] ) {
```